GoodDollar/GoodProtocol
# add: supertoken protocol nft upgrade + celo gas token

# Description
- Add compatibility with latest superfluid NFT upgrade
- Add debitGasFees and creditGasFees for Celo gas token compatability
- code size optimizations: removed unused functions, moved modifiers into inline function calls

See superfluid NFT upgrade:
https://github.com/superfluid-finance/protocol-monorepo/pull/1274
- NOTICE: superfluid added the deployment logic of the new NFTs inside the supertoken logic. Due to contract size issues, we assume that the NFTs will be deployed outside supergooddollar and they could be set using the setNFTProxyContracts method allowed only by the Avatar.
- only relevant contract in the PR is supertoken.sol we also had to change isupertoken.sol to remove unused methods.

See Celo instructions for gas token compatability:
- example implementations https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol
![image](https://user-images.githubusercontent.com/671095/227877486-2cfd57ba-5296-40d9-a841-2c88b7e6456a.png)
